### PR TITLE
fix(card-group): optional border for card group default

### PR DESCRIPTION
**Pull request by NSDrowned**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/6049

### Description

Include an optional border for `Cards group - Default` (React). 

*This branch included all commits from the web components version since they share the styles.

### CODE REVIEW

**Comment by DragosRistici** [comment]
Looking great for the Card group default! A few little details:

1. Should the Web Components and the React stories have to be different, or is it still a WIP? 
Screenshot 2021-06-01 at 12 34 50
Screenshot 2021-06-01 at 12 34 43

2. Web Components – With Card in card story – grid mode narrow (16px):
Screenshot 2021-06-01 at 12 38 39
Screenshot 2021-06-01 at 12 38 32
There is a right border for the first 2 cards that is applied to the dds-card-group-item which has padding instead of margin so the border is applied to the padding spacing also. 



3. Web Components – With Card in card story – the collapsed grid has some spacing errors
Screenshot 2021-06-01 at 12 43 34

4. The same errors from 2 and 3 are visible for the `With Card in card and Image cards` story
Screenshot 2021-06-01 at 12 44 22

**Comment by NSDrowned** [comment]
@DragosRistici I fixed the card in card border problem. As for the story names, I can update them to be the same as web components. For the 1px margin, it looks like that's the way it has been (canary looks the same). Not sure if it would be a separate issue since this issue is for the default variation?

**Comment by DragosRistici** [comment]
@NSDrowned Thank you, Julio. I will open a separate issue for the others. The Card in card – default looks good.

**Review by DragosRistici — APPROVED** [review]
Looks good.

**Review by kennylam — COMMENTED** [review]
Thanks for this @NSDrowned! Just a few small comments.

File: packages/react/src/components/CardGroup/CardGroup.js [hunk 1]
```diff
@@ -130,12 +130,15 @@ const CardGroup = ({ cards, cta }) => {
...
  * @returns {*} CardGroup JSX objects
  */
-const _renderCards = (cards, containerRef, cta) => (
+const _renderCards = (cards, containerRef, cta, border) => (
   <div
     data-autoid={`${stablePrefix}--card-group`}
-    className={`${prefix}--card-group__cards__row ${prefix}--row--condensed`}
+    className={`${prefix}--card-group__cards__row ${
+      border ? `${prefix}--card-group--border` : `${prefix}--row--condensed`
+    }`}
```
[the remark this answers is not in the document]
Comment (NSDrowned) [reply packages/react/src/components/CardGroup/CardGroup.js]: Fixed!

**Review by kennylam — APPROVED** [review]
LGTM thanks @NSDrowned!

**Pull request merged**